src/network/clientpackethandler.cpp

```diff
@@ -979,7 +979,14 @@ void Client::handleCommand_SpawnParticleBatch(NetworkPacket *pkt)
 
 	while (particle_data.peek() != EOF) {
 		auto p = std::make_unique<ParticleParameters>();
-		p->deSerialize(particle_data, m_proto_ver);
+		u32 particle_data_len = readU32(particle_data);
```

Comment:
but will a single particle ever be larger than u16? the only way I can think of is some terrible hacks with `[png`